inversion-cosmosdb/src/main/java/io/inversion/cosmosdb/CosmosDb.java

```diff
@@ -151,7 +151,31 @@ public String upsertRow(Collection collection, Map<String, Object> row) throws A
             if (id == null) {
                 id = collection.encodeResourceKey(row);
                 if (id == null)
+                {
+                    // this is the first potential problem, where we couldn't get the primary index for some reason
+                    Index index = collection.getPrimaryIndex();
+                    if(index == null) {
+                        log.warn("INVERSION UNABLE TO DETERMINE PRIMARY INDEX");
+                        collection.getIndexes().forEach(cIndex -> {
+                            log.info("    {}", cIndex.toString());
+                        });
+                    }
+                    else {
+                        // this is the second potential problem, where we found the index but couldn't construct it
+                        // either it's the wrong index being chosen, or the row, by the time it gets here, doesn't
+                        // have the data required to construct the thing
+                        log.info("INVERSION USING INDEX [{]]", index.toString());
```

Comment:
I think you typed a "]" instead of a "}" in your message, and the toString on index is again extraneous.

Reply:
meep!